File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/EventAnimation.java
package net.booone.chromaleague.hud.parts;

import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.Frame;
import net.booone.razersdk.animation.IFrame;
import net.booone.razersdk.animation.SimpleFrame;

public class EventAnimation extends AnimatedFrame {
    public void addAnimation(IFrame frame) {
        while (frame.hasFrame()) {
            addAnimationFrame(frame);
        }
    }

    @Override
    public Frame getFrame() {
        if (hasFrame()) {
            return super.getFrame();
        }
        return new SimpleFrame().getFrame();
    }
}

File: src/hud/parts/EventAnimation.ts
import {AnimatedFrame} from '../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../razer-sdk/animation/Frame.js';
import {IFrame} from '../../razer-sdk/animation/IFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';

export class EventAnimation extends AnimatedFrame {
    addAnimation(frame: IFrame): void {
        while (frame.hasFrame()) {
            this.addAnimationFrame(frame);
        }
    }

    override pollFrame(): Frame {
        if (this.hasFrame()) {
            return super.pollFrame();
        }
        return new SimpleFrame().pollFrame();
    }
}
